src/Rector/Class_/StaticDataProviderClassMethodRector.php

```diff
@@ -108,4 +110,20 @@ public function refactor(Node $node): ?Node
 
         return null;
     }
+
+    private function skipMethod(ClassMethod $classMethod): bool
+    {
+        if ($classMethod->isStatic()) {
+            return true;
+        }
+
+        $variables = $this->betterNodeFinder->findInstanceOf($classMethod, Variable::class);
+        foreach ($variables as $variable) {
+            if ($this->nodeNameResolver->isName($variable, 'this')) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
This can be changed to `findFirst`

```php
return (bool) $this->betterNodeFinder->findFirst(
            $classMethod->stmts,
            fn (Node $node): bool => $node instanceof Variable && $this->nodeNameResolver->isName($node, 'this')
        );
```

Reply:
I don't think it makes it easier to read, but see yourself:

https://github.com/rectorphp/rector-phpunit/pull/171/commits/e5bec06bd9e5235a3409299f6465087503beae8a

Reply:
`findFirst()` is faster as it stop on found node.